vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailActivity.kt

```diff
@@ -32,25 +34,44 @@ import im.vector.app.core.platform.ToolbarConfigurable
 import im.vector.app.core.platform.VectorBaseActivity
 import im.vector.app.databinding.ActivityRoomDetailBinding
 import im.vector.app.features.home.room.breadcrumbs.BreadcrumbsFragment
+import im.vector.app.features.matrixto.MatrixToBottomSheet
+import im.vector.app.features.navigation.Navigator
 import im.vector.app.features.room.RequireActiveMembershipAction
 import im.vector.app.features.room.RequireActiveMembershipViewEvents
 import im.vector.app.features.room.RequireActiveMembershipViewModel
 import im.vector.app.features.room.RequireActiveMembershipViewState
 import im.vector.app.features.widgets.permissions.RoomWidgetPermissionViewModel
 import im.vector.app.features.widgets.permissions.RoomWidgetPermissionViewState
-
 import javax.inject.Inject
 
 class RoomDetailActivity :
         VectorBaseActivity<ActivityRoomDetailBinding>(),
         ToolbarConfigurable,
         RequireActiveMembershipViewModel.Factory,
-        RoomWidgetPermissionViewModel.Factory {
+        RoomWidgetPermissionViewModel.Factory,
+        MatrixToBottomSheet.InteractionListener {
 
     override fun getBinding(): ActivityRoomDetailBinding {
         return ActivityRoomDetailBinding.inflate(layoutInflater)
     }
 
+    private val fragmentLifecycleCallbacks = object : FragmentManager.FragmentLifecycleCallbacks() {
```

Comment:
Suggestion: could this be extracted and reused? 

maybe something like this~

```kotlin
class MatrixBottomSheetCallbacks(
        private val provider: () -> MatrixToBottomSheet.InteractionListener
) : FragmentManager.FragmentLifecycleCallbacks() {
    override fun onFragmentResumed(fm: FragmentManager, f: Fragment) {
        if (f is MatrixToBottomSheet) {
            f.interactionListener = provider()
        }
    }

    override fun onFragmentPaused(fm: FragmentManager, f: Fragment) {
        if (f is MatrixToBottomSheet) {
            f.interactionListener = null
        }
    }
}

// activity
private val bottomSheetCallbacks = MatrixBottomSheetCallbacks { this }
```

Reply:
We also use SharedViewModel to let Fragments communicate between each others or with their host Activity

Reply:
@ouchadam feel free to open a dedicated PR to implement your suggestion :)